### PR TITLE
Additional assertions for SLM restart tests

### PR DESCRIPTION
**Pull request by jakelandis**
Adds the ability to conditionally wipe SLM policies
and ensures that you can read SLM policies across restarts.

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java [hunk 1]
```diff
@@ -485,8 +494,10 @@ private void wipeCluster() throws Exception {
             waitForPendingRollupTasks();
         }
 
-        // Clean up SLM policies before trying to wipe snapshots so that no new ones get started by SLM after wiping
-        deleteAllSLMPolicies();
+        if(preserveSLMPoliciesUponCompletion() == false) {
```
Comment (dakrone) [inline test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java]: Super minor nit:

```suggestion
        if (preserveSLMPoliciesUponCompletion() == false) {
```

**Review by dakrone — APPROVED** [review]
LGTM left one super minor nit

File: test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java (hunk 1 above)
Comment (andreidan) [reply test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java]: would it make sense to include this in the checkstyle rules? (ie. `WhitespaceAround`)

Comment (jakelandis) [reply test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java]: Changes the checkstyle rules can be pretty invasive across the board (and we don't want one-off rules). I think @pugnascotia is looking into a single commit reformatting, maybe this rule should be considered for that effort ?

Comment (pugnascotia) [reply test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java]: The formatting effort also introduces checking of the formatting, so we won't need to have specific rules for this sort of this.

**Comment by jakelandis** [comment]
@elasticmachine update branch

**Comment by jakelandis** [comment]
@elasticmachine run elasticsearch-ci/1

**Comment by jakelandis** [comment]
ran into a merge conflict on the back port. I will need make minor adjustments to the code here for 7.x.

**Comment by ywelsch** [comment]
This has not been backported to 7.x, blocking other backports (see #51430)

**Comment by ywelsch** [comment]
I've backported part of this (all except FullClusterRestartIT) to 7.x and 7.6 because of #51430.

**Comment by ywelsch** [comment]
I've adapted the version labels here as this has not been backported.

**Pull request merged**